app/process_proposal.go

```diff
@@ -79,25 +79,63 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) abci.ResponsePr
 
 		tx, err := app.txConfig.TxDecoder()(malleatedTx.Tx)
 		if err != nil {
-			// we don't reject the block here because it is not a block validity
-			// rule that all transactions included in the block data are
-			// decodable
-			continue
+			return abci.ResponseProcessProposal{
+				Result: abci.ResponseProcessProposal_REJECT,
+			}
 		}
 
+		var checkedTx bool
+
 		for _, msg := range tx.GetMsgs() {
 			if sdk.MsgTypeURL(msg) != types.URLMsgPayForBlob {
 				continue
 			}
 
+			if !checkedTx {
+				err = tx.ValidateBasic()
+				if err != nil {
+					app.Logger().Error("Tx including MsgPayForBlob is invalid")
+					return abci.ResponseProcessProposal{
+						Result: abci.ResponseProcessProposal_REJECT,
+					}
+				}
+
+				response := app.BaseApp.CheckTx(abci.RequestCheckTx{
+					Tx:   rawTx,
+					Type: abci.CheckTxType_New,
+				})
+
+				if response.GasUsed > response.GasWanted {
+					return abci.ResponseProcessProposal{
+						Result: abci.ResponseProcessProposal_REJECT,
+					}
+				}
```

Comment:
let's only check the signature for now as state in the target issue. As a side note, I think manually checking for gas would be more explicit